driver-async/src/test/functional/com/mongodb/async/client/TransactionsTest.java

```diff
@@ -391,6 +392,22 @@ public void execute() {
                             final BsonDocument arguments = operation.getDocument("arguments", new BsonDocument());
                             assertNull(sessionsMap.get(arguments.getString("session").getValue()).getPinnedServerAddress());
                         }
+                    } else if (operationName.equals("assertSessionTransactionState")) {
+                        final BsonDocument arguments = operation.getDocument("arguments", new BsonDocument());
+                        ClientSession session = sessionsMap.get(arguments.getString("session").getValue());
+                        String state = arguments.getString("state").getValue();
+                        if (state.equals("starting") || state.equals("in_progress")) {
+                            assertTrue(session.hasActiveTransaction());
+                        } else {
+                            assertFalse(session.hasActiveTransaction());
+                        }
```

Comment:
Transaction state isn't accessible and I didn't want to add it to the API for two tests.  Also, we don't have a starting state - just IN, so I opted to just test hasActiveTransaction().